pymc/model.py

```diff
@@ -1163,16 +1194,34 @@ def set_data(
                             f"{new_length}, so new coord values for the {dname} dimension are required."
                         )
                 if isinstance(length_tensor, TensorConstant):
+                    # The dimension was fixed in length.
+                    # Resizing a data variable in this dimension would
+                    # definitely lead to shape problems.
                     raise ShapeError(
                         f"Resizing dimension '{dname}' is impossible, because "
-                        "a 'TensorConstant' stores its length. To be able "
+                        "a `TensorConstant` stores its length. To be able "
                         "to change the dimension length, pass `mutable=True` when "
                         "registering the dimension via `model.add_coord`, "
                         "or define it via a `pm.MutableData` variable."
                     )
-                else:
+                elif length_tensor.owner is not None:
+                    # The dimension was created from a model variable.
                     length_belongs_to = length_tensor.owner.inputs[0].owner.inputs[0]
```

Comment:
Can you give me some context on what this `length_belongs_to` does? I have been staring at it for a while now without success. Is it checking if it comes from a call to `variable.shape` or something? What is the graph above `length_tensor` that we are assuming?

Reply:
```suggestion
                    # Get a handle on the tensor from which this dimension length was
                    # obtained by doing subindexing on the shape as ain `.shape[i]`.
                    # Needed to check if it was another shared variable.
                    length_belongs_to = length_tensor.owner.inputs[0].owner.inputs[0]
```